trie/trie.go

```diff
@@ -960,20 +982,23 @@ func (t *Trie) delete(origNode node, key []byte, keyStart int, preserveAccountNo
 	case *accountNode:
 		if keyStart >= len(key) || key[keyStart] == 16 {
 			// Key terminates here
+			h := key[:keyStart]
+			if h[len(h)-1] == 16 {
+				h = h[:len(h)-1]
+			}
 			if n.storage != nil {
-				h := key[:keyStart]
```

Comment:
The idea was not not to do the work unless we need to touch

Reply:
I now need it for `preserveAccountNode` as well and I didn't want to copy-paste